packages/manager/modules/overthebox/src/overthebox/autoconfigure/dns/template.html

```diff
@@ -0,0 +1,249 @@
+        </div>
+        <form
+            novalidate
+            name="formLocalDomain"
+            data-ng-if="$ctrl.showAddLocalDomain"
+        >
+            <div class="oui-input-group oui-input-group_inline">
+                <oui-field
+                    label="{{:: 'overTheBox_autoconfigure_hostname' | translate}}"
+                >
+                    <input
+                        class="oui-input"
+                        name="hostname"
+                        required
+                        ng-model="$ctrl.localDomain.hostname"
+                    />
+                </oui-field>
+                <oui-field
+                    class="ml-2"
+                    label="{{:: 'overTheBox_autoconfigure_ip' | translate}}"
+                >
+                    <input
+                        class="oui-input"
+                        name="ip"
+                        required
+                        ng-model="$ctrl.localDomain.ip"
+                    />
+                </oui-field>
+                <oui-field
+                    class="ml-2"
+                    label="{{:: 'overTheBox_autoconfigure_priority' | translate}}"
+                >
+                    <input
+                        class="oui-input"
+                        name="localDomainPriority"
+                        type="number"
+                        required
+                        ng-model="$ctrl.localDomain.priority"
+                    />
+                </oui-field>
+                <oui-button
+                    class="ml-2 mb-2"
+                    variant="primary"
+                    type="submit"
+                    on-click="$ctrl.createLocalDomain()"
+                    data-disabled="!$ctrl.localDomain.hostname || !$ctrl.localDomain.ip || !$ctrl.localDomain.priority"
+                >
+                    <span
+                        data-translate="overTheBox_autoconfigure_submit"
+                    ></span>
+                </oui-button>
+                <oui-button
+                    class="ml-2 mb-2"
+                    variant="primary"
+                    on-click="$ctrl.cancelCreateLocalDomain()"
+                >
+                    <span
+                        data-translate="overTheBox_autoconfigure_cancel"
+                    ></span>
+                </oui-button>
+            </div>
+        </form>
+        <oui-button
+            variant="primary"
+            data-on-click="$ctrl.addLocalDomain()"
+            class="mb-2 d-flex justify-content-center"
+            data-ng-if="$ctrl.displayAddLocalDomainButton && $ctrl.localDomains.length === 0"
+        >
+            <span
+                data-translate="overTheBox_autoconfigure_dns_local_domain_add"
+            ></span>
+        </oui-button>
+        <oui-datagrid
+            page-size="5"
+            rows="$ctrl.localDomains"
+            data-ng-if="$ctrl.localDomains.length > 0"
+        >
```

Comment:
You're setting a button, which populates a datagrid on click and disappear. Then the datagrid has a button itself to add one more item?

Why not having an empty datagrid, then disabling the button while adding the local domain etc...?

Reply:
I don't want to display an empty datagrid, I prefer display a button to add a new configuration. When the configuration is not empty, we can add another configuration with the button on the datagrid.

Reply:
Ok but that implies potentially page resizing as the datagrid is a big component. The better should be to have the page size moving as less as possible when using the application, and avoid blinking elements in the page

Reply:
Yes it could, but in this case, it's not blinking like you said and I think it's more understandable for the user that he need to click on the "Add" button to set a new configuration. If I set a blank table, it's less clear.